### PR TITLE
You Can Now Drag People Into Deep Friers

### PR DESCRIPTION
**Pull request by Erisfiregamer1**
# Description

You can now drag entities like people into deep fryers. Yes, this will cook them. Yes, you can cannibalize your own dead crew.

---

# TODO

- [ ] Add a doafter
- [ ] Add a popup warning everyone of what you've done

---

<details><summary><h1>Media</h1></summary>
<p>

obs is broken part 295

</p>
</details>

---

# Changelog

:cl:
- add: You can now shove people into deep fryers by drag-dropping them onto it.

### CODE REVIEW

File: Content.Server/Nyanotrasen/Kitchen/EntitySystems/DeepFryerSystem.cs [hunk 1]
```diff
@@ -347,6 +356,15 @@ private void DeepFry(EntityUid uid, DeepFryerComponent component, EntityUid item
                 "Ginormous" => 50,
                 _ => 10
             } * component.SolutionSizeCoefficient);
+        } else {
+            oilToUse = 10; // Default until I figure out how in the 24 hells to get an entity's mass and just use that
```
Comment (VMSolidus) [inline Content.Server/Nyanotrasen/Kitchen/EntitySystems/DeepFryerSystem.cs:360]:
```suggestion
            oilToUse = TryComp<PhysicsComponent>(item, out var physicsComponent) ? physicsComponent.Mass : 10;
```

**Review by VMSolidus — CHANGES_REQUESTED** [review]

File: Content.Server/Nyanotrasen/Kitchen/EntitySystems/DeepFryerSystem.cs (hunk 1 above)
Comment (Erisfiregamer1) [reply Content.Server/Nyanotrasen/Kitchen/EntitySystems/DeepFryerSystem.cs:360]: ah, so *that's* how I do that

**Comment by Erisfiregamer1** [comment]
@VMSolidus your changes broke compiling (it compiled fine before then)

i am now going to deep fry my IDE

**Comment by Erisfiregamer1** [comment]
WHY DID IT CLOSE IT I HIT COMMENT

**Comment by DocNITE** [comment]
mmmm... Fried harpy :woozy_face:

**Comment by Erisfiregamer1** [comment]
Ready for review, I've added the needed import and it should build.

Needs a Surgery fix though as when the affected burns up they don't actually dust and instead turn into a torsoless abomination

**Comment by BlueHNT** [comment]
If you can deep-fry brains, why not deep-fry people huh? The court of judgement will view this as Prime Murder for sure!

File: Content.Shared/Nyanotrasen/Kitchen/SharedDeepfryerSystem.cs [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+        var storable = HasComp<ItemComponent>(entity);
+        if (!storable && !HasComp<BodyComponent>(entity))
+            return false;
+
+        if (TryComp<PhysicsComponent>(entity, out var physics) && (physics.CanCollide) || storable)
+            return true;
+        else
+            return false;
+
+    }
```
[the remark this answers is not in the document]
Comment (Erisfiregamer1) [reply Content.Shared/Nyanotrasen/Kitchen/SharedDeepfryerSystem.cs:36]: I'll keep that in mind!

**Comment by Erisfiregamer1** [comment]
Ready for merge pending next positive review- the neccesary Shitmed update was merged to make it function Correctly.

**Review by sleepyyapril — APPROVED** [review]

**Pull request merged**